.github/scripts/questbook-i18n.py

```diff
@@ -1,60 +1,119 @@
-EN_US_LANG_PATH = 'config/txloader/load/minecraft/lang/en_US.lang'
-DEFAULT_QUESTS_US_PATH = 'config/betterquesting/DefaultQuests-us.json'
+DEFAULT_QUESTS_BASE_DIR_PATH = Path('config/betterquesting/DefaultQuests')
+QUEST_LINES_ORDER_PATH = DEFAULT_QUESTS_BASE_DIR_PATH / 'QuestLinesOrder.txt'
+QUEST_LINES_DIR_PATH = DEFAULT_QUESTS_BASE_DIR_PATH / 'QuestLines'
+QUESTS_DIR_PATH = DEFAULT_QUESTS_BASE_DIR_PATH / 'Quests'
+EN_US_LANG_PATH = DEFAULT_QUESTS_BASE_DIR_PATH / 'en_US.lang'
 
+ID_LENGTH = 24
 
-class Thing(metaclass=ABCMeta):
-    def __init__(self, obj: dict, id_key: str, title: str, infix: str):
-        self.obj: dict = obj
-        self.id: str = obj[id_key]
-        self.name: str = obj['properties:10']['betterquesting:10']['name:8'].replace('\n', '%n')
-        self.desc: str = obj['properties:10']['betterquesting:10']['desc:8'].replace('\n', '%n')
-        self.title = title
-        self.name_key = f'gtnh.{infix}{self.id}.name'
-        self.desc_key = f'gtnh.{infix}{self.id}.desc'
+DIR_NAME_NO_QUEST_LINE = 'NoQuestLine'
+DIR_NAME_MULTIPLE_QUEST_LINE = 'MultipleQuestLine'
 
-    def update_properties(self):
-        self.obj['properties:10']['betterquesting:10']['name:8'] = self.name_key
-        self.obj['properties:10']['betterquesting:10']['desc:8'] = self.desc_key
 
-    def write_to_lang_file(self, lang_file: TextIO):
-        lang_file.writelines([
-            '\n',
-            f'# {self.title}.{self.id} - {self.name}\n',
-            f'{self.name_key}={self.name}\n',
-            f'{self.desc_key}={self.desc}\n',
-        ])
+class BQJson(metaclass=ABCMeta):
+    def __init__(self, path: Path):
+        with open(path) as fp:
+            self.obj = json.load(fp)
+        self.path = path
 
+    @property
+    def name(self) -> str:
+        return self.obj['properties:10']['betterquesting:10']['name:8']
 
-class Quest(Thing):
-    def __init__(self, obj: dict):
-        super().__init__(obj, 'questID:3', 'Quest', 'quest')
+    @property
+    def desc(self) -> str:
+        return self.obj['properties:10']['betterquesting:10']['desc:8']
 
+    @property
+    def desc_without_break_line(self) -> str:
+        return self.desc.replace('\n', '%n')
 
-class QuestLine(Thing):
-    def __init__(self, obj: dict):
-        super().__init__(obj, 'lineID:3', 'QuestLine', 'line')
+    @property
+    def id(self) -> str:
+        return self.path.name.removesuffix('.json')[-ID_LENGTH:]
+
+    @property
+    def short_id(self) -> str:
+        return self.id.rstrip('=')
+
+    @abstractmethod
+    def write_to_lang_file(self, fp: TextIO):
+        pass
+
+
+class QuestLine(BQJson):
+    @property
+    def quests(self):
+        return self.obj['quests:9']
+
+    def write_to_lang_file(self, fp: TextIO):
+        fp.write(
+            f'\n\n'
+            f'## Quest Line: {quest_line.name}\n'
+            f'betterquesting.questline.{quest_line.short_id}.name={quest_line.name}\n'
```

Comment:
It might be a good idea to apply the replacement logic here as well, just in case some quest names contain '%'

Reply:
done